cf/commands/application/restart_test.go

```diff
@@ -63,27 +63,62 @@ var _ = Describe("restart command", func() {
 
 	Context("when logged in, targeting a space, and an app name is provided", func() {
 		BeforeEach(func() {
-			requirementsFactory.Application = app
 			requirementsFactory.LoginSuccess = true
 			requirementsFactory.TargetedSpaceSuccess = true
+			appRepo.ReadReturns.App = app
 
 			stopper.ApplicationStopReturns(app, nil)
 		})
 
-		It("restarts the given app", func() {
-			runCommand("my-app")
+		Context("app name is provided", func() {
+			It("restarts the given app", func() {
+				runCommand("my-app")
 
-			application, orgName, spaceName := stopper.ApplicationStopArgsForCall(0)
-			Expect(application).To(Equal(app))
-			Expect(orgName).To(Equal(config.OrganizationFields().Name))
-			Expect(spaceName).To(Equal(config.SpaceFields().Name))
+				application, orgName, spaceName := stopper.ApplicationStopArgsForCall(0)
+				Expect(application).To(Equal(app))
+				Expect(orgName).To(Equal(config.OrganizationFields().Name))
+				Expect(spaceName).To(Equal(config.SpaceFields().Name))
 
-			application, orgName, spaceName = starter.ApplicationStartArgsForCall(0)
-			Expect(application).To(Equal(app))
-			Expect(orgName).To(Equal(config.OrganizationFields().Name))
-			Expect(spaceName).To(Equal(config.SpaceFields().Name))
+				application, orgName, spaceName = starter.ApplicationStartArgsForCall(0)
+				Expect(application).To(Equal(app))
+				Expect(orgName).To(Equal(config.OrganizationFields().Name))
+				Expect(spaceName).To(Equal(config.SpaceFields().Name))
+
+				Expect(appRepo.ReadArgs.Name).To(Equal("my-app"))
+			})
+		})
 
-			Expect(requirementsFactory.ApplicationName).To(Equal("my-app"))
+		Context("app name is not provided", func() {
```

Comment:
Should we have a test for app name not provided and no manifest or lack of app name in manifest?